tools/jenkins/run_distribution.sh

```diff
@@ -66,26 +66,77 @@ if [ "$platform" == "linux" ]; then
-    # PATH=$dir/bin:$PATH brew install openssl
-    # PATH=$dir/bin:$PATH brew tap grpc/grpc
-    # PATH=$dir/bin:$PATH brew install --without-python google-protobuf
-    # PATH=$dir/bin:$PATH brew install grpc
-    PATH=$dir/bin:$PATH brew list -l
+    # system installed homebrew, don't interfere
     brew list -l
+
+    # Set up temp directories for test installation of homebrew
+    brew_root=/tmp/homebrew-test-$language
+    rm -rf $brew_root
+    mkdir -p $brew_root
+    git clone https://github.com/Homebrew/homebrew.git $brew_root
+
+    # Install grpc via homebrew
+    #
+    # The temp $PATH env variable makes sure we are operating at the right copy of
+    # temp homebrew installation, and do not interfere with the system's main brew
+    # installation.
+    #
+    # It's a bit non-ideal as we duplicate most of homebrew-grpc/script/install
+    # logic here but as the Jenkins macos instance has some different user/root
+    # privileges so spelling things out here might be cleaner.
+    cd $brew_root
+    PATH=$brew_root/bin:$PATH brew tap homebrew/dupes
+    PATH=$brew_root/bin:$PATH brew install zlib
+    PATH=$brew_root/bin:$PATH brew install openssl
+    PATH=$brew_root/bin:$PATH brew tap grpc/grpc
+    PATH=$brew_root/bin:$PATH brew install --without-python google-protobuf
+    PATH=$brew_root/bin:$PATH brew install grpc
+    PATH=$brew_root/bin:$PATH brew list -l
+
+    # Install per-language modules/extensions on top of core grpc
+    #
+    # If a command below needs root access, the binary had been added to
+    # /etc/sudoers. This step needs to be repeated if we add more mac instances
+    # to our jenkins project.
+    #
+    # Examples (lines that needed to be added to /etc/sudoers):
+    # + Defaults        env_keep += "CFLAGS CXXFLAGS LDFLAGS enable_grpc"
+    # + jenkinsnode1 ALL=(ALL) NOPASSWD: /usr/bin/pecl, /usr/local/bin/pip,
+    # +   /usr/local/bin/npm
+    case $language in
+      *core*) ;;
+      *python*)
+        sudo CFLAGS=-I$brew_root/include LDFLAGS=-L$brew_root/lib pip install grpcio
+        pip list | grep grpc
+        echo 'y' | sudo pip uninstall grpcio
+        ;;
+      *nodejs*)
+        sudo CXXFLAGS=-I$brew_root/include LDFLAGS=-L$brew_root/lib npm install --global grpc
+        npm list -g | grep grpc
+        sudo npm uninstall --global grpc
+        ;;
+      *ruby*)
+        gem install grpc -- --with-grpc-dir=$brew_root
+        gem list | grep grpc
+        gem uninstall grpc
+        ;;
+      *php*)
+        sudo enable_grpc=$brew_root CFLAGS="-Wno-parentheses-equality" pecl install grpc-alpha
+        pecl list | grep grpc
+        sudo pecl uninstall grpc
+        ;;
+      *)
+        echo "Unsupported language $language"
+        exit 1
+        ;;
+    esac
+
+    # clean up
     cd ~/ 
-    rm -rf $dir
-    echo $PATH # TODO: for debug, can be removed later
-    brew list -l # TODO: for debug, can be removed later
+    rm -rf $brew_root
```

Comment:
I think this can be skipped if the test fails, so I would suggest also doing it before the test.